provider/datagen/src/transform/cldr/currency/mod.rs

```diff
@@ -53,29 +52,43 @@ fn which_currency_pattern(
         Err(icu_properties::PropertiesError::PropDataLoad(e)) => {
             return Err(e);
         }
-        Err(_) => unreachable!(),
+        Err(_) => unreachable!("load_for_general_category_group should only return PropDataLoad"),
     };
 
-    let right_letter_test_position = {
-        if currency_sign_ind < first_num_ind && currency_sign_ind < last_num_ind {
-            true
-        } else if currency_sign_ind > first_num_ind && currency_sign_ind > last_num_ind {
-            false
+    #[derive(PartialEq)]
+    enum PlaceHolderCharCloserToNumber {
+        First,
+        Last,
+    }
+
+    let char_closer_to_number = {
+        if currency_sign_index < first_num_index {
+            PlaceHolderCharCloserToNumber::Last
+        } else if currency_sign_index > last_num_index {
+            PlaceHolderCharCloserToNumber::First
```

Comment:
Suggestion: You can simplify the code a bit by writing this like

```rust
let char_closest_to_number = if currency_sign_index < first_num_index {
    place_holder.chars().last().unwrap()
} else if currency_sign_index > last_num_index {
    place_holder.chars().first().unwrap()
} else {
    return Err(...);
};
if letters_set.as_borrowed().contains(char_closest_to_number) {
    Ok(PatternSelection::StandardAlphaNextToNumber)
} else {
    Ok(PatternSelection::Standard)
}
```

Reply:
good idea